samples/bluetooth/subrating/src/main.c

```diff
@@ -0,0 +1,443 @@
+
+	/* Start testing when the GATT service is discovered */
+	test_ready = true;
+	k_sem_give(&test_ready_sem);
+}
+
+static void discovery_service_not_found(struct bt_conn *conn, void *context)
+{
+	printk("Service not found\n");
+}
+
+static void discovery_error(struct bt_conn *conn, int err, void *context)
+{
+	printk("Error while discovering GATT database: (err %d)\n", err);
+}
+
+struct bt_gatt_dm_cb discovery_cb = {
+	.completed         = discovery_complete,
+	.service_not_found = discovery_service_not_found,
+	.error_found       = discovery_error,
+};
+
+static void adv_start(void)
+{
+	int err;
+
+	struct bt_le_adv_param *adv_param =
+		BT_LE_ADV_PARAM(BT_LE_ADV_OPT_CONNECTABLE |
+				BT_LE_ADV_OPT_ONE_TIME,
+				BT_GAP_ADV_FAST_INT_MIN_2,
+				BT_GAP_ADV_FAST_INT_MAX_2,
+				NULL);
+
+	err = bt_le_adv_start(adv_param, ad, ARRAY_SIZE(ad), sd, ARRAY_SIZE(sd));
+	if (err) {
+		printk("Advertising failed to start (err %d)\n", err);
+		return;
+	}
+
+	printk("Advertising successfully started\n");
+}
+
+static void scan_start(void)
+{
+	int err;
+
+	err = bt_scan_start(BT_SCAN_TYPE_SCAN_PASSIVE);
+	if (err) {
+		printk("Starting scanning failed (err %d)\n", err);
+		return;
+	}
+
+	printk("Scanning successfully started\n");
+}
+
+static void connected(struct bt_conn *conn, uint8_t err)
+{
+	if (err) {
+		printk("Connection failed, err 0x%02x %s\n", err, bt_hci_err_to_str(err));
+		return;
+	}
+
+	default_conn = bt_conn_ref(conn);
+	err = bt_conn_get_info(default_conn, &conn_info);
+	if (err) {
+		printk("Getting conn info failed (err %d)\n", err);
+		return;
+	}
+
+	/* make sure we're not scanning or advertising */
+	if (conn_info.role == BT_CONN_ROLE_CENTRAL) {
+		bt_scan_stop();
+	} else {
+		bt_le_adv_stop();
+	}
+
+	printk("Connected as %s\n",
+	       conn_info.role == BT_CONN_ROLE_CENTRAL ? "central" : "peripheral");
+
+	err = bt_gatt_dm_start(default_conn, BT_UUID_LATENCY, &discovery_cb, &latency_client);
+	if (err) {
+		printk("Discover failed (err %d)\n", err);
+	}
+}
+
+static void disconnected(struct bt_conn *conn, uint8_t reason)
+{
+	printk("Disconnected, reason 0x%02x %s\n", reason, bt_hci_err_to_str(reason));
+
+	test_ready = false;
+
+	if (default_conn) {
+		bt_conn_unref(default_conn);
+		default_conn = NULL;
+	}
+
+	if (conn_info.role == BT_CONN_ROLE_CENTRAL) {
+		scan_start();
+	} else {
+		adv_start();
+	}
+}
+
+static void subrate_defaults_set(void)
+{
+	int err = 0;
+
+	const struct bt_conn_le_subrate_param params = {
+		/* Minimum subrate factor. */
+		.subrate_min = 1,
+		/* Maximum subrate factor. */
+		.subrate_max = 10,
+		/* Maximum Peripheral latency in units of subrated connection intervals. */
+		.max_latency = 10,
+		/* Minimum number of underlying connection events to remain active
+		 * after a packet containing a Link Layer PDU with a non-zero Length
+		 * field is sent or received.
+		 */
+		.continuation_number = 0,
+		/* Maximum supervision timeout allowed in requests by a peripheral. */
```

Comment:
```suggestion
		/* Maximum supervision timeout */
```

While it's probably not too important, this is somewhat misleading. One can think that by calling `subrate_defaults_set` they can control what is allowed by the peripheral for good.

However, when central host does `subrate_request` it essentially overrides the allowed values set by `subrate_defaults_set` thus changing what the peripheral is allowed the send for future.

It's similar with the comment on [line 261](https://github.com/nrfconnect/sdk-nrf/pull/16863/files#diff-c67ae0cd3ceb4b4f8546b01c045f2b3dbc8fc6dd9b556a3e73c9a245ad57cd9dR261). There we could perhaps say "Set up default subrating limits for new connections" or something like that.